src/gui/TextEditor.cpp

```diff
@@ -865,6 +866,12 @@ void TextEditor::repaintCursor() { repaintEditor(); }
  * Blink!
  */
 auto TextEditor::blinkCallback(TextEditor* te) -> gint {
+    if (!te->cursorBlink) {
+        te->cursorVisible = true;
+        te->repaintCursor();
```

Comment:
This is a very bad way of doing this. `repaintCursor()` forces a full page redraw which is very expensive and might lead to more than just the cursor flashing... 

This PR would need repaintCursor to be reimplemented to at least only redraw the area of the TextEditor or even better only the area of the cursor.

Reply:
Ok, I've reimplemented `repaintCursor` to only repaint the area of the current text editor.  Replacing `repaintEditor` is harder because in general we need to repaint more than current area (for example when the user deletes a character).

Just to make it clear, the first version of the PR did not increase the number of page redraws (the number is even reduced if cursor blinking is disabled).  The text editor did a page redraw every time the cursor blinked.